### ISSUE TITLE
Grayscale textures are broken in 4.2.0.2188

### ISSUE THREAD

**Issue by Doprez** [stride3d/stride#2376]
**Release Type**: Official Release

**Version**: 4.2.0.2188

**Platform(s)**: Windows

**Describe the bug**
When importing a grayscale file into the editor it will be completely broken. 

![image](https://github.com/stride3d/stride/assets/73259914/bff5479c-8c09-4dbc-aafb-28fa036388f6)

you can make it at least shows something if you uncheck all of the options in the feature panel but it just shows white noise.

![image](https://github.com/stride3d/stride/assets/73259914/e073c25f-3bb4-4093-902b-0606dae65baa)

This is how it looks in 4.2.0.2122:

![image](https://github.com/stride3d/stride/assets/73259914/6f69a573-11bf-4130-b547-41f2a45af28b)

**To Reproduce**
1. get a grayscale image 
![BrickWall02_roughness](https://github.com/stride3d/stride/assets/73259914/ad5088b6-ee8a-45ba-9f82-da7a76848374)

2. import it into the editor

**Expected behavior**
Textures should look good.

**Comment by Doprez** [comment]
Some added info this seems to have broken after 2149 as it works in that gamestudio version as expected.
https://github.com/stride3d/stride/compare/releases/4.2.0.2149...releases/4.2.0.2188

**Comment by Doprez** [comment]
`System.DivideByZeroException: 'Attempted to divide by zero.'`
```
This exception was originally thrown at this call stack:
    Stride.Graphics.PixelFormatExtensions.SizeInBits(Stride.Graphics.PixelFormat) in PixelFormatExtensions.cs
    Stride.TextureConverter.TexLibraries.DxtTexLib.ChangeDxtImageType(Stride.TextureConverter.TexLibraries.DxtTextureLibraryData, Stride.TextureConverter.DxtWrapper.DXGI_FORMAT) in DxtTexLib.cs
    Stride.TextureConverter.TexLibraries.DxtTexLib.Load(Stride.TextureConverter.TexImage, Stride.TextureConverter.Requests.LoadingRequest) in DxtTexLib.cs
    Stride.TextureConverter.TexLibraries.DxtTexLib.Execute(Stride.TextureConverter.TexImage, Stride.TextureConverter.IRequest) in DxtTexLib.cs
    Stride.TextureConverter.TextureTool.Load(Stride.TextureConverter.Requests.LoadingRequest) in TextureTool.cs
    Stride.TextureConverter.TextureTool.Load(string, bool) in TextureTool.cs
    Stride.Assets.Textures.TextureAssetCompiler.TextureConvertCommand.DoCommandOverride(Stride.Core.BuildEngine.ICommandContext) in TextureAssetCompiler.cs
    Stride.Core.BuildEngine.Command.DoCommand(Stride.Core.BuildEngine.ICommandContext) in Command.cs
    Stride.Core.BuildEngine.CommandBuildStep.StartCommand(Stride.Core.BuildEngine.IExecuteContext, Stride.Core.IO.ListStore<Stride.Core.BuildEngine.CommandResultEntry>, Stride.Core.BuildEngine.BuilderContext) in CommandBuildStep.cs
```

**Comment by Jklawreszuk** [comment]
Weird, Are being able to reproduce this bug before my PR related to DirectXTex? https://github.com/stride3d/stride/pull/2338

**Comment by Doprez** [comment]
> Weird, Are being able to reproduce this bug before my PR related to DirectXTex? #2338

It seems like it may be before that change.

I just pulled a headless commit before https://github.com/stride3d/stride/pull/2303 was merged and now it works.

**Comment by Jklawreszuk** [comment]
@Doprez Could you upload tested texture here in this issue? I wonder what gone wrong that it executes DxtTextLib in the first place...

**Comment by tebjan** [comment]
@Jklawreszuk it is the one in the post above, probably it has a high bit format. Maybe 16, 24 or 32 bit per channel: [Bricks]( https://private-user-images.githubusercontent.com/73259914/346364854-ad5088b6-ee8a-45ba-9f82-da7a76848374.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA1MjQzMjYsIm5iZiI6MTcyMDUyNDAyNiwicGF0aCI6Ii83MzI1OTkxNC8zNDYzNjQ4NTQtYWQ1MDg4YjYtZWU4YS00NWJhLTlmODItZGE3YTc2ODQ4Mzc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA5VDExMjAyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5NDY1NjY3MWUyM2Q1NzcyYWFjNDZhZDRjMWJiZDA1ZTQwZWYxMzYyYTllMGU3OGIzOWM2NTk4M2VhZDMwM2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Mbl0S4dnBQWAHUiCanrQeGcKfT84aTTHEzd21DJppOg)

**Comment by tebjan** [comment]
@Jklawreszuk what was the reason to change the supported file formats in #2303? The DxtLib handled png files with esoteric pixel formats correctly.

**Comment by Doprez** [comment]
> @Doprez Could you upload tested texture here in this issue? I wonder what gone wrong that it executes DxtTextLib in the first place...

It should be the image in the description, unfortunately it shows the picture instead of just a file reference but opening athat and downloading it should reproduce the issue.

**Comment by Jklawreszuk** [comment]
@Doprez Right, thanks! I didn't pay attention to it 🤦 
@tebjan LoadFromWICFile depends on Win32 which is available on Windows only, that's why was gotten removed. 

DirectXTex wiki : https://github.com/microsoft/DirectXTex/wiki/WIC-I-O-Functions.

So intention was to move responsibility of image loading from dxt to freeimage but I guess modifying `SupportedExtensions` variable isn't enough. Will try this week to see what needs to be changed additionally to make the import work properly. If I don't find any idea to fix, restoring the LoadFromWICFile method while adding a workaround for Linux systems may also be some solution to the problem 😅

**Comment by Basewq** [comment]
Not urgent for me, but is there any progress on this issue?

From what I can tell, DxtTexLib loads the images in whatever pixel format and only changes DxtTexLib would make is converting srgb/non-srgb:
https://github.com/stride3d/stride/blob/6a2e29b6aa4aeaf852631f9466b05b201d182142/sources/tools/Stride.TextureConverter/Backend/TexLibraries/DxtTexLib.cs#L247
eg. That 16-bit grayscale image gets loaded as a 16-bit pixel format texture.

FITexLib, on the other hand would try to force the loaded image as 32-bit pixel format (BGRA_8888):
https://github.com/stride3d/stride/blob/dad92aaa2bd294ae302261fdde52583ab329a6f5/sources/tools/Stride.TextureConverter/Backend/TexLibraries/FITexLib.cs#L246
The 16-bit grayscale image above just fails to get converted (ie. returns as null data), and it basically falls apart from then on.

The solution seems to be either update FITexLib to somehow retain the original pixelformat, or just revert it until someone knows how to fix it through FITexLib (not me).

I haven't tested to see, but I am also concerned that `Execute` method in DxtTexLib has things that FITexLib doesn't do (MipMapsGeneration, NormalMapGeneration). Not sure if FITexLib not implementing those means it'll get sorted out somewhere else or not.

**Comment by Eideren** [comment]
Closed in favor of the more specific #2449
